### PR TITLE
reopened server show more detail error msg

### PR DESCRIPTION
**Pull request by zhangwinning**
#### Checklist
Fix https://github.com/fastify/fastify/issues/2411

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by zhangwinning** [comment]
cc @mcollina, thanks

File: test/close.test.js [hunk 1]
```diff
@@ -233,3 +233,37 @@ t.test('Current opened connection should not accept new incoming connections', t
...
+    .then(() => {
+      fastify.close()
+        .then(() => {
+          fastify.listen(0).catch(err => {
+            t.ok(err)
+            t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
+          })
+        })
+    })
+})
```
Comment (mcollina) [inline test/close.test.js:251]: would you mind to use async/await in the test?

File: test/close.test.js [hunk 2]
```diff
@@ -233,3 +233,37 @@ t.test('Current opened connection should not accept new incoming connections', t
...
+    t.error(err)
+
+    fastify.close((err) => {
+      t.error(err)
+
+      fastify.listen(0, err => {
+        t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
+        t.ok(err)
+      })
+    })
```
Comment (mcollina) [inline test/close.test.js:267]: would you mind to use async/await in the test?

Comment (zhangwinning) [reply test/close.test.js:267]: thanks @mcollina. 
I try use async/await, while when I change below it,  show error for it `tapError: incorrect number of tests`,  plan test is 2, I dig it for a while, but not solved. please told me where is errors. 
```

test('Cannot be reopened the closed server has listen callback', async t => {
  t.plan(2)
  const fastify = Fastify()

  await fastify.listen(0)
  await fastify.close()

  fastify.listen(0, (err) => {
    t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
    t.ok(err)
  })
})
```

Comment (Eomm) [reply test/close.test.js:267]: you must write

```
try{
  fastify.listen(0)
}catch(err){
    t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
    t.ok(err)
}
```

Comment (zhangwinning) [reply test/close.test.js:267]: Hey @Eomm thanks for the answer.

The test unit is `listen has callback `.  if I follow  the code about 👆，I don't test code logic. https://github.com/fastify/fastify/pull/2415/files#diff-c945a46d13b34fcaff544d966cffcabaR138.

Comment (Eomm) [reply test/close.test.js:267]: Ops 😅

Then you can just wrap the callback in a new Promise 👍🏼

Comment (zhangwinning) [reply test/close.test.js:262]: thanks @Eomm, I got it

File: test/close.test.js (hunk 1 above)
Comment (zhangwinning) [reply test/close.test.js:250]: I got it

**Pull request merged**